Source/Core/sampleTerrain.js

```diff
@@ -78,23 +150,14 @@ function doSampling(terrainProvider, level, positions) {
     tileRequestSet[key].positions.push(positions[i]);
   }
 
-  // Send request for each required tile
   const tilePromises = [];
-  for (i = 0; i < tileRequests.length; ++i) {
-    const tileRequest = tileRequests[i];
-    const requestPromise = tileRequest.terrainProvider.requestTileGeometry(
-      tileRequest.x,
-      tileRequest.y,
-      tileRequest.level
-    );
-    const tilePromise = requestPromise
-      .then(createInterpolateFunction(tileRequest))
-      .catch(createMarkFailedFunction(tileRequest));
-    tilePromises.push(tilePromise);
-  }
-
-  return Promise.all(tilePromises).then(function () {
-    return positions;
+  log("starting to drain request queue");
+  return drainTileRequestQueue(tileRequests, tilePromises).then(function () {
+    log("all requests made, waiting for all responses");
+    return Promise.all(tilePromises).then(function () {
+      log("all inflight requests are complete, sampling done");
```

Comment:
Can we remove the logging in this class? It's useful for debugging, but won't be useful to the end user.